### PR TITLE
Revive & rework AI status displays

### PR DESCRIPTION
**Pull request by BatElite**
[feat][rework][bug]
## About the PR 
This PR revives the AI status displays that were mothballed early in April (a commit somewhere in #109 of all places). There's also a few random AI-related fixes because you can't do a medium-sized PR without finding broken things.

The displays have been reworked a bit:

-The sprite is 1px thinner and taller The screen now fits unedited AI faces and the sprite is somewhat in perspective.
-Displays now use equivalent face overlays as AI units themselves, so they can be coloured like AIs can and they give off some light.
-Text and malfunctioning faces are ported over properly. Sadly these more than compensate for the dmi space the previous 
-Displays now have an owner variable, so that a second AI being activated doesn't override all the displays. They also name themselves after whichever AI they're owned by. 
-Rather than sending data over like they used to, AI code mostly just handles claiming ownership of displays (they grab everything free on login) and turning them on and off. The display's process code checks its owner's state and updates accordingly.
-Material requirements are explicit now (2 metal, 6 crystal & conductive)
-Displays now offset upwards, so they can be deployed on the floor in front of walls.

In order to not have to change every map in this PR to add some displays (and predicting people have opinions on where they should go), I've included a temporary solution by adding a blueprint in mechanic's lockers. Those should be removed once most or all current maps are updated with displays of their own. The only map touched is Donut 3, which already had 2 displays on the AI sat perimeter for some reason and which had to be moved down a tile.

Because I'm only one person I've not quite been able to test multiple AIs interacting WRT ownership. It seems to all be working though.

Miscelaneous changes & fixes
-Fixes a runtime in the malfunctioning AI skin if you redeemed it as an AI eye (I thought I needed to poke that reward for the displays but didn't)
-The AI blank face has a new sprite that's just the blank face plate instead of the entire damn body, which could otherwise layer over damage overlays.
-AIs now dispose of their light properly.

## Why's this needed? 
These displays are pretty cool and they should come back if possible. Gives AIs a more direct presence on station (provided mechanics get on with making them). Going rogue is nicer if you can fill the halls with red laughing skulls.

(Honestly I'm not really seeing why they were removed in the first place. I was told they were really laggy, but the code looked fairly tame and the POS texticon proc that was being blamed belonged to a different object.)

afbeelding

## Changelog

```changelog
(u)BatElite
(*)Engineers have found some dusty AI displays in storage, and upgraded them to better represent out beloved silicon friends! Until NT stations and ships are properly fitted with the displays, mechanics can find some blueprints for them in their lockers.
(+)Each AI status display now has an "owner", so it's possible to share them between AIs. You can still claim/steal any screen from other AIs by clicking on them.
```

### CODE REVIEW

File: code/obj/machinery/status_display.dm [hunk 1]
```diff
@@ -368,71 +372,125 @@
...
+				back_image.color = face_color
+				UpdateOverlays(back_image, "back_img")
+				UpdateOverlays(glow_image, "glow_img", 1) //forced so it displays on top
+				UpdateOverlays(face_image, "emotion_img", 1) //idem
+
+				var/colors = GetColors(face_color)
+				screen_glow.set_color(colors[1] / 255, colors[2] / 255, colors[3] / 255)
+
+			message = owner.status_message
+			name = initial(name) + " ([owner.name])"
```
Comment (BatElite) [inline code/obj/machinery/status_display.dm:457]: For these two lines I thought that checking if message/name changed would be about as expensive in lookup stuff as just setting them, so you'd win nothing if you don't have to update and lose out if you do. I don't actually know if that's accurate though.

File: code/obj/machinery/manufacturer.dm [hunk 2]
```diff
@@ -1922,6 +1922,12 @@
 	blueprint = /datum/manufacture/mechanics/loafer
 
 
+/******************** AI Display Blueprints (should be temporary but we know how that goes in coding) *******************/
```
Comment (comradef191) [inline code/obj/machinery/manufacturer.dm:1925]: i can try to figure out how to checkout your branch and add them all back in if ya want

File: code/obj/machinery/status_display.dm [hunk 3]
```diff
@@ -368,71 +372,125 @@
...
+		UpdateOverlays(face_image, "emotion_img")
+
+	get_desc()
+		..()
+		if (status & NOPOWER)
+			return
+		if (src.message)
+			. += "<br>[owner.name] says: \"[src.message]\""
+
+	verb/claim() //Captain said it's my turn on the status display
```
Comment (pali6) [inline code/obj/machinery/status_display.dm:478]: I think AI could just claim the display by clicking on it instead? (Done thourgh overriding `attack_ai()`.)

Comment (BatElite) [reply code/obj/machinery/status_display.dm:478]: The more you know

**Comment by ZeWaka** [comment]
>Text and malfunctioning faces are ported over properly.

This doesn't sound like it uses maptext then...

File: code/datums/manufacturing.dm [hunk 4]
```diff
@@ -115,6 +115,16 @@ proc/get_nice_mat_name_for_manufacturers(mat)
 	create = 1
 	frame_path = /obj/disposalpipe/loafer
 
+/******************** AI display (temp) *******************/
+
+/datum/manufacture/mechanics/ai_status_display
+	name = "AI display"
+	item_paths = list("MET-1","CON-1","CRY-1")
+	item_amounts = list(2,6,6)
```
Comment (pali6) [inline code/datums/manufacturing.dm:123]:
```suggestion
```
You can now just leave this out and it'll use `mats`.

File: code/mob/living/silicon/ai.dm [hunk 5]
```diff
@@ -1234,11 +1236,28 @@ var/list/ai_emotions = list("Happy" = "ai_happy",\
 	..()
 	update_clothing()
 	src.updateOverlaysClient(src.client) //ov1
+	if (!isdead(src))
+		for (var/obj/machinery/ai_status_display/O in machine_registry[MACHINES_STATUSDISPLAYS]) //change status
+			SPAWN_DBG(0)
```
Comment (pali6) [inline code/mob/living/silicon/ai.dm:1241]: I don't see why the spawn is necessary here. (But also be sure to change the `return` to `continue` if you decide to remove the spawn.)

File: code/mob/living/silicon/ai.dm [hunk 6]
```diff
@@ -1234,11 +1236,28 @@ var/list/ai_emotions = list("Happy" = "ai_happy",\
...
+				O.owner = src
+				O.mode = 1
 	return
 
 /mob/living/silicon/ai/Logout()
 	src.removeOverlaysClient(src.client) //ov1
 	..()
+	// Only turn off the status displays if we're dead.
+	if (isdead(src))
+		SPAWN_DBG(0)
```
Comment (pali6) [inline code/mob/living/silicon/ai.dm:1253]: Again I don't quite see the necessity of the spawn.

**Comment by BatElite** [comment]
> This doesn't sound like it uses maptext then...

Should maybe have clarified, but text here refers to the AI skin . It's an animation.
These indeed don't use maptext, but they're not trying to draw sentences.

---

WRT the SPAWNS I copied them in from the old code, and then forgot to ask why they were there (one is inside the for, another out, quality code :P)
Will remove them and see if it keeps working.

**Comment by BatElite** [comment]
The fuck did it include a bunch of other people's PRs for

File: strings/changelog.txt [hunk 7]
```diff
@@ -1,4 +1,9 @@
 
+(t)sat jan 30 21
+(u)zjdtmkhzt
+(p)3509
+(e)⚖🆕|balance, feature
+(+)Artifact activators are now more common, but not all of them will be universal anymore. Be wary of those of eldritch origin...
```
Comment (pali6) [inline strings/changelog.txt:6]: remove this pls

File: code/obj/artifacts/faults.dm [hunk 8]
```diff
@@ -111,6 +111,7 @@
 		elecflash(user,power = 6, exclude_center = 0)
 		user.stuttering += 30
 
+ABSTRACT_TYPE(/datum/artifact_fault/messager/)
```
Comment (pali6) [inline code/obj/artifacts/faults.dm:114]: and this

File: code/obj/artifacts/faults.dm [hunk 9]
```diff
@@ -1,5 +1,5 @@
 // FAULTS
-
+ABSTRACT_TYPE(/datum/artifact_fault/)
```
Comment (pali6) [inline code/obj/artifacts/faults.dm:2]: and this

File: code/obj/artifacts/artifactprocs.dm [hunk 10]
```diff
@@ -199,12 +199,15 @@
...
 
 		if (K.activated)
-			if (ACT.universal || A.artitype == K.artitype)
-				if (ACT.activator && !A.activated)
+			if (K.universal || A.artitype == K.artitype)
+				if (K.activator && !A.activated)
 					src.ArtifactActivated()
+					if(K.corrupting && A.faults.len < 10) // there's only so much corrupting you can do ok
+						for(var/i=1,i<rand(1,3),i++)
+							src.ArtifactDevelopFault(100)
```
Comment (pali6) [inline code/obj/artifacts/artifactprocs.dm:210]: etc

**Comment by BatElite** [comment]
Having irreperably fucked over this branch I've PRed the same thing as a fresh branch. Sorry for all that.

**Pull request closed without merging**